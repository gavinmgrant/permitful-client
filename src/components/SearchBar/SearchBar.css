.search {
    width: 100%;
    background: none;
}

.search-input {
    position: fixed;
    margin-left: 10px;
    margin-right: 150px;
    margin-top: 10px;
    left: 0;
    right: 0;
    z-index: 10;
}

.search input,
.limit-input input {
    padding: 0.5rem;
    height: 50px;
    border: 2px solid #18181E;
    border-radius: 25px;
}

.limit-input {
    position: fixed;
    margin-left: auto;
    margin-right: 10px;
    width: 130px;
    margin-top: 10px;
    left: 0;
    right: 0;
    z-index: 10;
}

.to-results {
    position: fixed;
    margin-left: auto;
    margin-right: auto;
    margin-top: 40px;
    left: 0;
    right: 0;
    z-index: 10;
}

[id^=listbox] {
    color: #18181E;
}

@media screen and (min-width: 1024px) {
    .search-input {
        margin-left: auto;
        margin-right: auto;
        width: 800px;
        right: 170px;
    }

    .limit-input {
        margin-right: auto;
        left: 840px;
    }
}