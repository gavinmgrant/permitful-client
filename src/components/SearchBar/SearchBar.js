import React, { useState, useEffect, useContext } from 'react';
import PermitfulContext from '../../contexts/PermitfulContext';
import usePlacesAutoComplete, { getGeocode, getLatLng } from "use-places-autocomplete";
import { Combobox, ComboboxInput, ComboboxPopover, ComboboxList, ComboboxOption } from '@reach/combobox';
import Results from '../Results/Results';
import '@reach/combobox/styles.css';
import './SearchBar.css'

export default function SearchBar(props) {
  const context = useContext(PermitfulContext);

  // determine the geographic limits of the search bar
  let cityLat, cityLng;
  if (context.cityName === 'SFO') {
    cityLat = 37.773972;
    cityLng = -122.431297;
  } else if (context.cityName === 'LAX') {
    cityLat = 34.0522;
    cityLng = -118.2437;
  };
  const { ready, value, suggestions: {status, data}, setValue, clearSuggestions } = usePlacesAutoComplete({
    requestOptions: {
      location: { lat: () => cityLat, lng: () => cityLng },
      radius: 200 * 1000,
    },
  });

  const panTo = props.getPanTo;

  // sends search address to parent, permit map
  const [searchAddress, setSearchAddress] = useState('');
  const handleAdd = props.handleAddress;
  useEffect(() => {
    handleAdd(searchAddress);
  }, [searchAddress, handleAdd]);

  // sends marker limit amount to parent, permit map
  const [markerLimit, setMarkerLimit] = useState(100);
  const handleLimit = props.handleMarkerLimit;
  useEffect(() => {
    handleLimit(markerLimit);
  }, [markerLimit, handleLimit]);

  return (
    <div className="search">
      <Combobox 
        onSelect={async (address) => {
          setValue(address, false);
          clearSuggestions();

          try {
            const results = await getGeocode({address});
            const { lat, lng } = await getLatLng(results[0]);
            panTo({ lat, lng });
            setSearchAddress(address);
          } catch(error) {
            console.log("Combobox error!");
          }
        }}
      >
        <div className="search-input">
          <ComboboxInput 
            value={value} 
            onChange={(e) => {
              setValue(e.target.value);
            }} 
            disabled={!ready}
            placeholder="Search by address"
          />
          <ComboboxPopover>
            <ComboboxList>
              {status === "OK" && 
                data.map(({ description }) => (
                  <ComboboxOption key={description} value={description} /> 
                ))}
            </ComboboxList>
          </ComboboxPopover>
        </div>
      </Combobox> 
      <div className="limit-input">
        <input 
          type="number" 
          id="limit" 
          name="limit" 
          placeholder={'# of markers'}
          min="0"
          max="10000"
          onChange={e => {
            setMarkerLimit(e.target.value);
          }}
        /> 
      </div>
      {!searchAddress ? <div></div> :
        <div>
          <Results searchAddress = {searchAddress}/>
        </div> 
      }
    </div>
  ); 
}