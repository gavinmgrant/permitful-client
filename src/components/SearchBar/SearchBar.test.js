import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBar from './SearchBar';

describe(`SearchBar component`, () => {
    it('renders a search bar form by default', () => {
      const wrapper = shallow(<SearchBar />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
});