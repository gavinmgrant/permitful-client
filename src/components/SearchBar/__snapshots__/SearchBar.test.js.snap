// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders a search bar form by default 1`] = `
<div
  className="search"
>
  <Combobox
    onSelect={[Function]}
  >
    <div
      className="search-input"
    >
      <ComboboxInput
        disabled={true}
        onChange={[Function]}
        placeholder="Search by SF address"
        value=""
      />
      <ComboboxPopover>
        <ComboboxList />
      </ComboboxPopover>
    </div>
  </Combobox>
  <div
    className="limit-input"
  >
    <input
      id="limit"
      max="10000"
      min="0"
      name="limit"
      onChange={[Function]}
      placeholder="# of markers"
      type="number"
    />
  </div>
  <div />
</div>
`;
